pkg/ntpmonitor/ntpmonitor_test.go

```diff
@@ -99,3 +129,141 @@ func TestNewFromConfig(t *testing.T) {
 		t.Errorf("unexpected error %s", err)
 	}
 }
+
+func TestNTPMonitorQueryNTPServer(t *testing.T) {
+	mockNTP := MockNTPClient{}
+	failNTP := MockNTPClient{
+		ignoredServers: map[string]string{
+			"s1": "",
+		},
+	}
+
+	testCases := []struct {
+		name             string
+		client           MockNTPClient
+		expectTestToPass bool
+	}{
+		{
+			name:             "Successfully query NTP server",
+			client:           mockNTP,
+			expectTestToPass: true,
+		},
+		{
+			name:             "Fail to query NTP server",
+			client:           failNTP,
+			expectTestToPass: false,
+		},
+	}
+	for _, tc := range testCases {
+		monitor, err := NewFromConfig(&Config{
+			Servers:         []string{"s1"},
+			NumServers:      1,
+			RequestAttempts: 1,
+			ServerThreshold: 1,
+			RequestTimeout:  1,
+			MaxTimeDelta:    1,
+		})
+		if err != nil {
+			t.Fatalf("unexpectedly failed to create NTP monitor: %v", err)
+		}
+		monitor.WithCustomClient(tc.client)
+
+		resp, err := monitor.QueryNTPServer("s1")
+		if tc.expectTestToPass && err != nil {
+			t.Errorf("test '%s' unexpectedly failed with non-nil error: %v", tc.name, err)
+		}
+		if tc.expectTestToPass && resp == nil {
+			t.Errorf("test '%s' unexpectedly failed with nil ntp.Response", tc.name)
+		}
+		if !tc.expectTestToPass && err == nil {
+			t.Errorf("test '%s' unexpectedly passed with a nil error", tc.name)
+		}
```

Comment:
Can we also test metrics output? (I'm not sure!)

Reply:
I'll look into this.